File: python/src/aegis/resources/jailbreak.py
"""Jailbreak resource — V2 jailbreak detection endpoints."""

from __future__ import annotations

from typing import Any, Dict, List

from aegis.resources._base import AsyncResource, SyncResource


def _to_text(item: Any) -> str:
    if isinstance(item, str):
        return item
    if isinstance(item, dict):
        return item.get("text") or item.get("content") or ""
    return getattr(item, "text", None) or getattr(item, "content", None) or ""


class SyncJailbreak(SyncResource):

    def detect(self, content: str, **kwargs: Any) -> Dict[str, Any]:
        return self._post("/v2/jailbreak/detect", json={"text": content, **kwargs})

    def detect_batch(self, requests: List[Any]) -> Dict[str, Any]:
        return self._post(
            "/v2/jailbreak/detect/batch",
            json={"texts": [_to_text(r) for r in requests]},
        )

    def types(self) -> List[Dict[str, Any]]:
        data = self._get("/v2/jailbreak/types")
        return data if isinstance(data, list) else data.get("types", [])


class AsyncJailbreak(AsyncResource):

    async def detect(self, content: str, **kwargs: Any) -> Dict[str, Any]:
        return await self._post("/v2/jailbreak/detect", json={"text": content, **kwargs})

    async def detect_batch(self, requests: List[Any]) -> Dict[str, Any]:
        return await self._post(
            "/v2/jailbreak/detect/batch",
            json={"texts": [_to_text(r) for r in requests]},
        )

    async def types(self) -> List[Dict[str, Any]]:
        data = await self._get("/v2/jailbreak/types")
        return data if isinstance(data, list) else data.get("types", [])

File: go/jailbreak.go
package aegis

import (
	"context"

	"github.com/ai-aegis-yatav/aegis-sdk/go/internal"
)

// JailbreakService — V2 jailbreak detection.
type JailbreakService struct {
	t *internal.Transport
}

func (s *JailbreakService) Detect(ctx context.Context, text string) (map[string]any, error) {
	var resp map[string]any
	if err := s.t.Do(ctx, "POST", "/v2/jailbreak/detect", map[string]any{"text": text}, &resp); err != nil {
		return nil, err
	}
	return resp, nil
}

func (s *JailbreakService) DetectBatch(ctx context.Context, texts []string) (map[string]any, error) {
	var resp map[string]any
	if err := s.t.Do(ctx, "POST", "/v2/jailbreak/detect/batch", map[string]any{"texts": texts}, &resp); err != nil {
		return nil, err
	}
	return resp, nil
}

func (s *JailbreakService) Types(ctx context.Context) (any, error) {
	var resp any
	if err := s.t.Do(ctx, "GET", "/v2/jailbreak/types", nil, &resp); err != nil {
		return nil, err
	}
	return resp, nil
}
